test/namedtuple.jl

```diff
@@ -21,6 +21,13 @@
 @test (a=3,)[:a] == 3
 @test (x=4, y=5, z=6).y == 5
 @test (x=4, y=5, z=6).z == 6
+@test (x=4, y=5, z=6)[(:x, :y)] == (x=4, y=5)
+@test (x=4, y=5, z=6)[(:x,)] == (x=4,)
+# If the following are included, is it worth putting in 1 line with above, eg:
+# @test (x=4, y=5, z=6)[(:x, :y)] == (x=4, y=5, z=6)[[:x, :y]] == (x=4, y=5)
+# or keep separate, eg:
+@test (x=4, y=5, z=6)[[:x, :y]] == (x=4, y=5)
+@test (x=4, y=5, z=6)[[:x]] == (x=4,)
```

Comment:
Could add some `@test_throws` on stuff like `(a=1,)[(:x,)]` or `(x=1, y=2)[(:x,:x)]` also corner cases like `(a=...)[()]` and `NamedTuple()[()]` could be added.

Reply:
Hmm - good thinking. For corner cases, what are the expected behaviors? For the later, I think I might expect it to return another empty tuple, since 

```
julia> [][:]
Any[]
```

But I don't know. And for the first of your corner cases, I'm assuming you mean `(a=...)` as shorthand for an actual named tuple? Or did you mean something using splatting?

Reply:
Yes, I meant e.g. `@test (a=1, b=2)[()] === NamedTuple()` and in general `@test nt[()] === NamedTuple()`